src/routes/join/components/SignupPage.js

```diff
@@ -0,0 +1,180 @@
+const React = require('react');
+const T = require('prop-types');
+const { default: Styled } = require('styled-components');
+const { default: Typography } = require('@material-ui/core/Typography');
+const { default: TextField } = require('@material-ui/core/TextField');
+const { default: Button } = require('@material-ui/core/Button');
+const { default: Box } = require('@material-ui/core/Box');
+const { default: Link } = require('@material-ui/core/Link');
+const { default: FormControl } = require('@material-ui/core/FormControl');
+const { default: InputLabel } = require('@material-ui/core/InputLabel');
+const { default: Input } = require('@material-ui/core/Input');
+const { default: InputAdornment } = require('@material-ui/core/InputAdornment');
+const { default: IconButton } = require('@material-ui/core/IconButton');
+const { default: Visibility } = require('@material-ui/icons/Visibility');
+const { default: VisibilityOff } = require('@material-ui/icons/VisibilityOff');
+const StrangeForms = require('strange-forms');
+
+const internals = {};
+
+module.exports = class SignupPage extends StrangeForms(React.Component) {
+
+    static propTypes = {
+        reqCreateAccount: T.func.isRequired
+    };
+
+    static fields = {
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: ''
+    };
+
+    defaultSubmit = (ev) => {
+
+        ev.preventDefault();
+        this.handleNext();
+    }
+
+    constructor(props) {
+
+        super(props);
+
+        this.state = {
+            firstName: '',
+            lastName: '',
+            email: '',
+            password: '',
+            showPassword: false,
+            isSubmitting: false
+        };
+
+        this.strangeForm({
+            fields: Object.keys(SignupPage.fields),
+            get: (someProps, field) => {
+
+                return someProps.data ? someProps.data[field] : '';
+            },
+            act: () => null
+        });
+    }
+
+    handleClickShowPassword = (ev) => {
+
+        this.setState((currentState) => ({
+            showPassword: !currentState.showPassword
+        }));
+    }
+
+    disableSubmit() {
+
+        return false;
+    }
+
+    handleSubmit = async (ev) => {
+
+        ev.preventDefault();
+        const accountInfo = this.formatFields();
+        this.setState({ isSubmitting: true });
+        const [err, result] = await this.props.reqCreateAccount(accountInfo);
+        this.setState({ isSubmitting: false });
+        if (!err && result.success) {
```

Comment:
I think this `result.success` stuff can go, since it looks like it catered to some other API than user-pal.
```suggestion
        if (!err) {
```

Reply:
should we cut [result] from the destructuring too?

Reply:
If you don't need it then yeah, I'd say so!  I think the linter might even give you a little warning about the unused variable.